### ISSUE TITLE
Misaligned address error

### ISSUE THREAD

**Issue by Xandaros** [OpenVR-Advanced-Settings/OpenVR-AdvancedSettings#559]
**Describe the bug**
AdvancedSettings immediately exits with `SIGBUS (Misaligned address error)`

**To Reproduce**
1. Open AdvancedSettings

**Software (please complete the following information):**
 - OS: Arch Linux
 - Advanced Settings Version: 565d25ea15eecf2f287d19c482b087ae012ada5e

**Hardware (please complete the following information):**
 - CPU: AMD Ryzen 5 5600X
 - Graphics Card: NVIDIA GTX 970
 - HMD: Valve Index

**Additional context**
I've first tried the version available through steam, which just quits immediately. Downloading the latest released AppImage results in the aforementioned SIGBUS:
```
xandaros@MELON tmp> ./OpenVR_Advanced_Settings-d5cade5-STEAM-x86_64.AppImage
[INFO] 2021-03-15 23:44:16: Application started (Version 5.3.2)
[INFO] 2021-03-15 23:44:16: Log File: /home/xandaros/.local/share/AdvancedSettings-Team/AdvancedSettings.log
[INFO] 2021-03-15 23:44:16: Settings File: /home/xandaros/.config/AdvancedSettings-Team/OVR Advanced Settings.ini
[INFO] 2021-03-15 23:44:16: defaultlockXToggle: 'false' | lockYToggle: 'false' | lockZToggle: 'false' | momentumSave: 'false' | turnBindLeft: 'false' | turnBindRight: 'false' | turnBounds: 'false' | moveShortcutLeft: 'false' | moveShortcutRight: 'false' | dragBounds: 'false' | allowExternalEdits: 'false' | oldStyleMotion: 'false' | universeCenteredRotation: 'false' | enableSeatedMotion: 'false' | adjustChaperone: 'true' | showLogMatricesButton: 'false' | simpleRecenter: 'false' | adjustChaperone2: 'false' | disableVersionCheck: 'false' | previousShutdownSafe: 'true' | vsyncDisabled: 'false' | crashRecoveryDisabled: 'false' | enableDebug: 'false' | enableExclusiveInput: 'false' | pttEnabled: 'false' | pttShowNotification: 'false' | micProximitySensorCanMute: 'false' | micReversePtt: 'false' | alarmEnabled: 'false' | alarmIsModal: 'true' | vrcDebug: 'false' | brightnessEnabled: 'false' | isOverlayMethodActive: 'false' | colorOverlayEnabled: 'false' | chaperoneSwitchToBeginnerEnabled: 'false' | chaperoneHapticFeedbackEnabled: 'false' | chaperoneAlarmSoundEnabled: 'false' | chaperoneAlarmSoundLooping: 'true' | chaperoneAlarmSoundAdjustVolume: 'false' | chaperoneShowDashboardEnabled: 'false' | disableChaperone: 'false' | centerMarkerNew: 'false' | autoturnEnabled: 'false' | autoturnUseCornerAngle: 'true' | autoturnVestibularMotionEnabled: 'false' | autoturnViewRatchettingEnabled: 'false' | autoturnShowNotification: 'true' | defaultheightToggleOffset: '-1.000000' | gravityStrength: '9.800000' | flingStrength: '1.000000' | brightnessOpacityValue: '1.000000' | colorOverlayOpacity: '0.000000' | colorRedNew: '1.000000' | colorGreenNew: '1.000000' | colorBlueNew: '1.000000' | chaperoneSwitchToBeginnerDistance: '0.500000' | chaperoneHapticFeedbackDistance: '0.500000' | chaperoneAlarmSoundDistance: '0.500000' | chaperoneShowDashboardDistance: '0.500000' | fadeDistanceRemembered: '0.500000' | activationDistance: '0.400000' | deactivateDistance: '0.150000' | cordDetanglingAngle: '0.261799' | autoturnMinCordTangle: '6.283185' | autoturnVestibularMotionRadius: '22.000000' | autoturnViewRatchettingPercent: '0.050000' | defaultkeyboardOne: '^>m' | keyboardTwo: '^>m' | keyboardThree: '^>m' | keyPressmisc: 'F9' | keyPressSystem: 'F9' | defaultsnapTurnAngle: '4500' | smoothTurnRate: '100' | dragComfortFactor: '0' | turnComfortFactor: '0' | frictionPercent: '0' | debugState: '0' | customTickRateMs: '20' | alarmHour: '0' | alarmMinute: '0' | alarmSecond: '0' | autoturnLinearTurnSpeed: '45000' | autoturnMode: '1' |
[INFO] 2021-03-15 23:44:16: Command line arguments processed.
fish: “./OpenVR_Advanced_Settings-d5ca…” terminated by signal SIGBUS (Misaligned address error)
```

I then tried compiling from source. Same problem.
I went all the way back to a29e6baaf7f591ab4c9e71d72a58baae8be1d22b, but still the same issue.
I doubt such a problem went undetected for over a year, so it's probably not the codebase, but I honestly have no idea what could be causing this.

I also ran AdvancedSettings through gdb and got a backtrace after the crash:
```
#0  0x00007ffff7ff0153 in memset () from /lib64/ld-linux-x86-64.so.2
#1  0x00007ffff7fd440a in _dl_map_object_from_fd () from /lib64/ld-linux-x86-64.so.2
#2  0x00007ffff7fd582d in _dl_map_object () from /lib64/ld-linux-x86-64.so.2
#3  0x00007ffff7fda2f5 in openaux () from /lib64/ld-linux-x86-64.so.2
#4  0x00007ffff5726b38 in _dl_catch_exception () from /usr/lib/libc.so.6
#5  0x00007ffff7fda683 in _dl_map_object_deps () from /lib64/ld-linux-x86-64.so.2
#6  0x00007ffff7fdffd4 in dl_open_worker () from /lib64/ld-linux-x86-64.so.2
#7  0x00007ffff5726b38 in _dl_catch_exception () from /usr/lib/libc.so.6
#8  0x00007ffff7fdfade in _dl_open () from /lib64/ld-linux-x86-64.so.2
#9  0x00007ffff54fb34c in ?? () from /usr/lib/libdl.so.2
#10 0x00007ffff5726b38 in _dl_catch_exception () from /usr/lib/libc.so.6
#11 0x00007ffff5726c03 in _dl_catch_error () from /usr/lib/libc.so.6
#12 0x00007ffff54fbb89 in ?? () from /usr/lib/libdl.so.2
#13 0x00007ffff54fb3d8 in dlopen () from /usr/lib/libdl.so.2
#14 0x00007ffff7cf6d0f in SharedLib_Load (pchPath=<optimized out>) at /data/src/public/vrsdk/src/sharedlibtools_public.cpp:18
#15 0x00007ffff7cf1b7f in vr::VR_LoadHmdSystemInternal () at /data/src/public/vrsdk/src/openvr_api_public.cpp:146
#16 0x00007ffff7cf1df9 in vr::VR_InitInternal2 (peError=0x7fffffffb7b0, eApplicationType=vr::VRApplication_Overlay, pStartupInfo=0x0) at /data/src/public/vrsdk/src/openvr_api_public.cpp:60
#17 0x00005555555b397a in openvr_init::initializeProperly(openvr_init::OpenVrInitializationType) ()
#18 0x00005555555b3d09 in openvr_init::initializeOpenVR(openvr_init::OpenVrInitializationType) ()
#19 0x0000555555570317 in main ()
```
My experience with GDB is rather limited, so this is about as much debugging as I can do with it, unfortunately.

**Comment by username223** [comment]
Thank you for a literally perfect issue ticket.

The problem happens in the OpenVR init code. `openvr_init::initializeProperly` is our code and `vr::VR_InitInternal2` is Valve code.

The call to Valve code happens [here](https://github.com/OpenVR-Advanced-Settings/OpenVR-AdvancedSettings/blob/11c0231a6a1838fd97f7ba3aa60f54526be6c7a9/src/openvr/openvr_init.cpp#L23). The call takes two parameters and they're both well formed, so I doubt that it's our code causing the problem.

OpenVR calls (external Valve code) `VR_LoadHmdSystemInternal` [here](https://github.com/ValveSoftware/openvr/blob/4c85abcb7f7f1f02adaf3812018c99fc593bc341/src/openvr_api_public.cpp#L60). The `SharedLib_Load` call occurs [here](https://github.com/ValveSoftware/openvr/blob/4c85abcb7f7f1f02adaf3812018c99fc593bc341/src/openvr_api_public.cpp#L147) with `"vrclient.so"` as a C string parameter. `dlopen` is called [here](https://github.com/ValveSoftware/openvr/blob/4c85abcb7f7f1f02adaf3812018c99fc593bc341/src/vrcommon/sharedlibtools_public.cpp#L20) with what appears to be well formed parameters.

Since nothing weird is going on before the call to `dlopen` there's either a bug in `libdl` (unlikely) or there's something configuration related going on.

--------

Could you try:

1. Ensuring that SteamVR is running and functional before trying to open OVRAS. Windows versions of OVRAS can force SteamVR to run, but this doesn't seem to work on Linux.
2. All the "hidden" dependencies of SteamVR are met. I mainly had issues with graphics stuff like `libgl` and `libvulkan`, but there may be others.

My `vrclient.so` is at `~/.local/share/Steam/steamapps/common/SteamVR/bin` and running `ldd` on it gives the following:

```
	linux-gate.so.1 (0xf7fc7000)
	libSDL2-2.0.so.0 => /lib/i386-linux-gnu/libSDL2-2.0.so.0 (0xf77c7000)
	libGL.so.1 => /lib/i386-linux-gnu/libGL.so.1 (0xf775a000)
	librt.so.1 => /lib/i386-linux-gnu/librt.so.1 (0xf774f000)
	libdl.so.2 => /lib/i386-linux-gnu/libdl.so.2 (0xf7749000)
	/lib/ld-linux.so.2 (0xf7fc9000)
	libpthread.so.0 => /lib/i386-linux-gnu/libpthread.so.0 (0xf7727000)
	libm.so.6 => /lib/i386-linux-gnu/libm.so.6 (0xf7620000)
	libc.so.6 => /lib/i386-linux-gnu/libc.so.6 (0xf7429000)
	libasound.so.2 => /lib/i386-linux-gnu/libasound.so.2 (0xf7313000)
	libpulse.so.0 => /lib/i386-linux-gnu/libpulse.so.0 (0xf72b4000)
	libX11.so.6 => /lib/i386-linux-gnu/libX11.so.6 (0xf7163000)
	libXext.so.6 => /lib/i386-linux-gnu/libXext.so.6 (0xf714a000)
	libXcursor.so.1 => /lib/i386-linux-gnu/libXcursor.so.1 (0xf713d000)
	libXinerama.so.1 => /lib/i386-linux-gnu/libXinerama.so.1 (0xf7138000)
	libXi.so.6 => /lib/i386-linux-gnu/libXi.so.6 (0xf7123000)
	libXrandr.so.2 => /lib/i386-linux-gnu/libXrandr.so.2 (0xf7114000)
	libXss.so.1 => /lib/i386-linux-gnu/libXss.so.1 (0xf710f000)
	libXxf86vm.so.1 => /lib/i386-linux-gnu/libXxf86vm.so.1 (0xf7107000)
	libdrm.so.2 => /lib/i386-linux-gnu/libdrm.so.2 (0xf70f1000)
	libgbm.so.1 => /lib/i386-linux-gnu/libgbm.so.1 (0xf70e0000)
	libwayland-egl.so.1 => /lib/i386-linux-gnu/libwayland-egl.so.1 (0xf70db000)
	libwayland-client.so.0 => /lib/i386-linux-gnu/libwayland-client.so.0 (0xf70cb000)
	libwayland-cursor.so.0 => /lib/i386-linux-gnu/libwayland-cursor.so.0 (0xf70bf000)
	libxkbcommon.so.0 => /lib/i386-linux-gnu/libxkbcommon.so.0 (0xf7078000)
	libGLdispatch.so.0 => /lib/i386-linux-gnu/libGLdispatch.so.0 (0xf6ffb000)
	libGLX.so.0 => /lib/i386-linux-gnu/libGLX.so.0 (0xf6fbf000)
	libpulsecommon-13.99.so => /usr/lib/i386-linux-gnu/pulseaudio/libpulsecommon-13.99.so (0xf6f32000)
	libdbus-1.so.3 => /lib/i386-linux-gnu/libdbus-1.so.3 (0xf6ed2000)
	libxcb.so.1 => /lib/i386-linux-gnu/libxcb.so.1 (0xf6ea3000)
	libXrender.so.1 => /lib/i386-linux-gnu/libXrender.so.1 (0xf6e97000)
	libXfixes.so.3 => /lib/i386-linux-gnu/libXfixes.so.3 (0xf6e8f000)
	libwayland-server.so.0 => /lib/i386-linux-gnu/libwayland-server.so.0 (0xf6e78000)
	libexpat.so.1 => /lib/i386-linux-gnu/libexpat.so.1 (0xf6e4c000)
	libffi.so.8 => /lib/i386-linux-gnu/libffi.so.8 (0xf6e42000)
	libsystemd.so.0 => /lib/i386-linux-gnu/libsystemd.so.0 (0xf6d81000)
	libwrap.so.0 => /lib/i386-linux-gnu/libwrap.so.0 (0xf6d75000)
	libsndfile.so.1 => /lib/i386-linux-gnu/libsndfile.so.1 (0xf6ce3000)
	libasyncns.so.0 => /lib/i386-linux-gnu/libasyncns.so.0 (0xf6cdc000)
	libapparmor.so.1 => /lib/i386-linux-gnu/libapparmor.so.1 (0xf6cc7000)
	libXau.so.6 => /lib/i386-linux-gnu/libXau.so.6 (0xf6cc1000)
	libXdmcp.so.6 => /lib/i386-linux-gnu/libXdmcp.so.6 (0xf6cb9000)
	liblzma.so.5 => /lib/i386-linux-gnu/liblzma.so.5 (0xf6c8b000)
	libzstd.so.1 => /lib/i386-linux-gnu/libzstd.so.1 (0xf6bc0000)
	liblz4.so.1 => /lib/i386-linux-gnu/liblz4.so.1 (0xf6b9d000)
	libgcrypt.so.20 => /lib/i386-linux-gnu/libgcrypt.so.20 (0xf6ab9000)
	libnsl.so.1 => /lib/i386-linux-gnu/libnsl.so.1 (0xf6a9c000)
	libFLAC.so.8 => /lib/i386-linux-gnu/libFLAC.so.8 (0xf6a61000)
	libogg.so.0 => /lib/i386-linux-gnu/libogg.so.0 (0xf6a55000)
	libvorbis.so.0 => /lib/i386-linux-gnu/libvorbis.so.0 (0xf6a29000)
	libvorbisenc.so.2 => /lib/i386-linux-gnu/libvorbisenc.so.2 (0xf699c000)
	libresolv.so.2 => /lib/i386-linux-gnu/libresolv.so.2 (0xf6982000)
	libbsd.so.0 => /lib/i386-linux-gnu/libbsd.so.0 (0xf6961000)
	libgpg-error.so.0 => /lib/i386-linux-gnu/libgpg-error.so.0 (0xf6938000)

```

**Comment by Xandaros** [comment]
SteamVR was definitely running and working. I actually first attempted to open AdvancedSettings from within VR. I never closed it during testing and went on to play a VR game after opening the issue. I think it is safe to assume that SteamVR was running and in working condition for the duration.

The ldd output on my vrclient.so (in `~/.steam/steamapps/common/SteamVR/bin`) is a LOT shorter:
```
	linux-gate.so.1 (0xf7ed1000)
	libSDL2-2.0.so.0 => /usr/lib32/libSDL2-2.0.so.0 (0xf769c000)
	libGL.so.1 => /usr/lib32/libGL.so.1 (0xf7630000)
	librt.so.1 => /usr/lib32/librt.so.1 (0xf7625000)
	libdl.so.2 => /usr/lib32/libdl.so.2 (0xf761f000)
	/usr/lib/ld-linux.so.2 (0xf7ed3000)
	libpthread.so.0 => /usr/lib32/libpthread.so.0 (0xf75fd000)
	libm.so.6 => /usr/lib32/libm.so.6 (0xf752f000)
	libc.so.6 => /usr/lib32/libc.so.6 (0xf7337000)
	libGLdispatch.so.0 => /usr/lib32/libGLdispatch.so.0 (0xf72bb000)
	libGLX.so.0 => /usr/lib32/libGLX.so.0 (0xf7280000)
	libX11.so.6 => /usr/lib32/libX11.so.6 (0xf7131000)
	libxcb.so.1 => /usr/lib32/libxcb.so.1 (0xf7103000)
	libXau.so.6 => /usr/lib32/libXau.so.6 (0xf70fe000)
	libXdmcp.so.6 => /usr/lib32/libXdmcp.so.6 (0xf70f6000)
```
None of them are unresolved, but the difference in length does give me pause. I opted out of the SteamVR beta, which I was using and verified the integrity of SteamVR. ldd output is unchanged and OVRAS still does not run.
For completeness, I also tried the `linux_v1.14` version of SteamVR, but to no avail.

Since it seemingly fails during initialisation, I figured I'd try running some other OpenVR dashboard, but don't really know of any. I tried to build the official samples, but unfortunately can't seem to build them.
I don't think it is related, but in case it is, the build fails with `/usr/bin/ld: error: /usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../lib/libSDL2.so: file too short`. The file mentioned is a symlink to a 1.5M, so I'm not sure what the problem is.

**Comment by username223** [comment]
> I tried to build the official samples, but unfortunately can't seem to build them.

In the past I've been unable to build OpenVR for both Windows and Linux for weird reasons, so this is not a unique case.

I don't know of any third party dashboards that currently support Linux. If you find one then you should definitely try it out.

I'm suspecting that it's an OpenVR issue or config issue, so if you could confirm this by replicating the error on another dashboard that would be awesome.

When you compiled from source did you use the git repo or the AUR?

**Comment by Xandaros** [comment]
Straight from the git repo. I followed [these instructions](https://github.com/OpenVR-Advanced-Settings/OpenVR-AdvancedSettings/blob/master/docs/building_for_linux.md#archmanjaro-4).
As I mentioned, I also tried some older versions of OVRAS. Some of them required some changes to the Makefile to be able to be built, but for the current version I followed the instructions almost verbatim. (I used `-j12` instead of `-j2`. So... not really any changes)
